pykrige/uk.py

```diff
@@ -281,13 +297,26 @@ def __init__(self, x, y, z, variogram_model='linear', variogram_parameters=None,
                 self.variogram_function = variogram_function
         else:
             self.variogram_function = self.variogram_dict[self.variogram_model]
+
         if self.verbose:
             print("Initializing variogram model...")
+
+        # see comment in ok.py about 'use_psill' kwarg...
+        if variogram_parameters is None:
+            vp_temp = None
+        elif use_psill:
+            vp_temp = variogram_parameters
+        else:
+            if self.variogram_model in ['gaussian', 'spherical', 'exponential', 'hole-effect']:
+                vp_temp = [variogram_parameters[0] - variogram_parameters[2],
+                           variogram_parameters[1], variogram_parameters[2]]

```

Comment:
Good idea, having a separate private function will reduce redundancy (what we're going for, after all). My goal here was provide a means for the user to provide either the full sill or the partial sill. I've changed things to work with the partial sill, as that fixes some of the annoying problems with the automatic variogram estimation routine. So if the user specifies the full sill, needed some way to convert to a parameter list that contains the partial sill...